presto-main/src/main/java/com/facebook/presto/operator/PagesIndex.java

```diff
@@ -513,12 +513,28 @@ public LookupSourceSupplier createLookupSourceSupplier(
                 session,
                 hashStrategy,
                 valueAddresses,
-                channels,
+                toPages(channels),
                 filterFunctionFactory,
                 sortChannel,
                 searchFunctionFactories);
     }
 
+    private static List<Page> toPages(List<List<Block>> channels)
```

Comment:
Is there a good place to place this as a public static function so we can share the implementation with `PagesRTreeIndex`?